lib/rummager/app.rb

```diff
@@ -274,6 +274,38 @@ def get_type_from_request_body(request_body)
     simple_json_result(action)
   end
 
+  get "/ltr/train" do
+    require_authentication "ltr_training"
+    halt(403, "LTR is not enabled") unless Search::RelevanceHelpers.ltr_enabled?
+
+    status = LearnToRank::DataPipeline.get_status
+    code = case status
+           when LearnToRank::DataPipeline::READY_STATUS
+             200
+           when LearnToRank::DataPipeline::ERROR_STATUS
+             500
```

Comment:
I'm not sure we should return a 500 here. A 200 would probably be more accurate for successful status requests, since the request succeeds. We can add more details about the status of the object/job within the JSON response, e.g.

```json
200
{ "job": { "status": { "code": "FAILED", "message": "The job failed" } } }
```

Reply:
I'm also not sure what the appropriate error to return here is but, by that reasoning, all three cases should return 200 - the status got successfully returned.  I wanted to use different status codes so that whatever is consuming this API doesn't need to recognise the error text we return.